draft-ietf-oauth-selective-disclosure-jwt.md

```diff
@@ -537,447 +539,170 @@ Note: Examples in this document use `cnf` Claim defined in [@RFC7800] to include
-      1. Determine that the public key for the private key that used to sign the HS-Disclosures JWT is bound to the SD-JWT, i.e., the SD-JWT either contains a reference to the public key or contains the public key itself.
-      2. Determine that the HS-Disclosures JWT is bound to the current transaction and was created for this Verifier (replay protection). This is usually achieved by a `nonce` and `aud` field within the HS-Disclosures JWT.
-    2. For each claim in `sd_hs_disclosures` in the HS-Disclosures JWT:
-      3. Ensure that the claim is present as well in `sd_digests` in the SD-JWT.
-         If `sd_digests` is structured, the claim MUST be present at the same
-         place within the structure.
-      4. Compute the base64url-encoded digest of the JSON literal disclosed
-         by the Holder using the `sd_digest_derivation_alg` in SD-JWT.
-      5. Compare the digests computed in the previous step with the one of
-         the same claim in the SD-JWT. Accept the claim only when the two
-         digests match.
-      6. Ensure that the claim value in the HS-Disclosures JWT is a JSON-encoded
-         object containing at least the keys `s` and `v`, and optionally `n`.
-      7. Store the value of the key `v` as the claim value. If `n` is contained
-         in the object, use the value of the key `n` as the claim name.
-    3. Once all necessary claims have been verified, their values can be
-       validated and used according to the requirements of the application. It
-       MUST be ensured that all claims required for the application have been
-       disclosed.
-
-If any step fails, the input is not valid and processing MUST be aborted.
-
-## Processing Model {#processing_model}
-
-Neither an SD-JWT nor an HS-Disclosures JWT is suitable for direct use by an application.
-Besides the REQUIRED verification steps listed above, it is further RECOMMENDED
-that an application-consumable format is generated from the data released in
-the HS-Disclosures. The RECOMMENDED way is to merge the released claims and any
-plaintext claims in the SD-JWT recursively:
-
- * Objects from the released claims must be merged into existing objects from the SD-JWT.
- * If a key is present in both objects:
-   * If the value in the released claims is an object and the value in the
-     SD-JWT claims is an object, the two objects MUST be merged recursively.
-   * Else, the value in the released claims MUST be used.
-
-The keys `sd_digests` and `sd_digest_derivation_alg` SHOULD be removed prior to further
-processing.
-
-The processing is shown in Examples 2b and 3 in the Appendix.
+    5. Check that the `sd_digest_derivation_alg` claim is present and its value is understood and the digest derivation algorithm is deemed secure.
+ 4. Create a copy of the SD-JWT payload for further processing, if required.
```

Comment:
why?

Reply:
I wasn't sure if we need to make this explicit, but the next steps are most easily described by saying how the JWT needs to be "modified". If we don't say that you can do this on a copy, people might complain when they have the JWT in a data structure that cannot be modified.

Reply:
```suggestion
 4. Create a copy of the SD-JWT payload, if required for further processing.
```